### PR TITLE
Fixes typos in holonomic.py, singularityfunction.py and diophantine.py

### PR DESCRIPTION
**Pull request by Abdullahjavednesar**
Typos in holonomic singularityfunction and diophantine

### CODE REVIEW

**Comment by Abdullahjavednesar** [comment]
@jksuom, @asmeurer I was looking into these files, and came across some possible typographical errors, and I do feel there are many more which needs attention. May be better raising an issue to update them.

File: sympy/holonomic/holonomic.py [hunk 1]
```diff
@@ -1231,7 +1231,7 @@ def to_sequence(self, lb=True):
         Each tuple in this vector represents a recurrence relation :math:`R`
         associated with a root of the indicial equation ``p``. Conditions of
         a different format can also be provided in this case, see the
-        docstring of HolonomicFunction class.
+        docstring of Holonomic Function class.
```
Comment (jksuom) [inline sympy/holonomic/holonomic.py]: `HolonomicFunction` is the name of a class. It should be left in that form.

Comment (Abdullahjavednesar) [inline sympy/holonomic/holonomic.py]: Opps! I'm sorry, that's quite obvious.

**Comment by jksuom** [comment]
Thanks!

**Comment by Abdullahjavednesar** [comment]
Pleasure always!

**Pull request merged**